src/raw.rs

```diff
@@ -210,10 +228,15 @@ where
     /// Creates a `RawTask` from a raw task pointer.
     #[inline]
     pub(crate) fn from_ptr(ptr: *const ()) -> Self {
+        // TODO: This function technically should be unsafe, since ptr must point to a region that
+        // has a size and alignment matching task layout, since doing pointer arithmetic that
+        // leaves the region or creating unaligned pointers is UB.
```

Comment:
Good point. When this function was written this was safe, but now with pointer provenance it is no longer safe. Could you mark this as `unsafe`?

Reply:
Done